dlib/control/policy.h

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2017 Adrián Javaloy (adrian.javaloy@gmail.com)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_POLICY_Hh_
+#define DLIB_POLICY_Hh_
+
+#include "../matrix.h"
+#include "policy_abstract.h"
+#include <iostream>
+#include <random>
+
+namespace dlib
+{
+
+    template <
+        typename model_type
+        >
+    class greedy_policy
```

Comment:
Rename this to policy rather than having the alias statement later on.  This is the default policy people will use in most cases right?

Reply:
Also, put the policy objects back into approximate_linear_models.h.  Save moving things around to another PR since moving stuff around makes the review much harder.  I can't tell what things are just moves vs substantive code changes.

Also, per my other comment, I'm not sure it's a good idea to move these things outside approximate_linear_models_abstract.h anyway.